src/components/common/Dialogs/ControlSafeDialog/SafeTransactionPreview.tsx

```diff
@@ -0,0 +1,454 @@
+  /* onlyForceAction, */
+  canManageAndControlSafes,
+}: Props) => {
+  const {
+    watch,
+    setValue,
+    formState: { isSubmitting },
+  } = useFormContext();
+
+  const safe = watch('safe');
+  const transactions: SafeTransaction[] = watch(`transactions`);
+
+  const [transactionTabStatus, setTransactionTabStatus] = useState(
+    Array(transactions.length).fill(false),
+  );
+  const { formatMessage } = useIntl();
+
+  const tokens = useMemo(
+    () => transactions.map((t) => t.tokenData),
+    [transactions],
+  );
+
+  const handleTabToggle = (newIndex: number) => {
+    const newTransactionTabs = transactionTabStatus.map((tab, index) =>
+      index === newIndex ? !tab : tab,
+    );
+    setTransactionTabStatus([...newTransactionTabs]);
+  };
+
+  const getTransactionTypeLabel = (transactionTypeValue: string) => {
+    const transactionType = transactionOptions.find(
+      (transaction) => transaction.value === transactionTypeValue,
+    );
+    return transactionType ? formatMessage(transactionType.label) : null;
+  };
+
+  const autogeneratedIds = useMemo(
+    () => [...new Array(transactions.length)].map(nanoid),
+    [transactions.length],
+  );
+
+  const isNFT = (index) =>
+    transactions[index].transactionType === TransactionTypes.TRANSFER_NFT;
+
+  /*
+   * Remove unused contract functions from form state.
+   * Doing it here instead of in the Contract Interaction section so that the user doesn't lose state in the
+   * event they switch between contract functions. We need this so the correct functions appear on the actions page.
+   */
+  useEffect(() => {
+    transactions.forEach((transaction, index) => {
+      const actualSelectedFunction = transaction.contractFunction;
+      const allSelectedMethodKeys = Object.keys(
+        omit(transaction, Object.keys(defaultTransaction)),
+      );
+      const keysToExclude = allSelectedMethodKeys.filter(
+        // the keys end with "-[functionName]", so we exclude the ones that don't end in the
+        // function name that the user ended up choosing
+        (key) => !new RegExp(`-${actualSelectedFunction}$`).test(key),
+      );
+      const updatedTransaction = {
+        ...omit(transaction, keysToExclude),
+      };
+      setValue(`transactions.${index}`, updatedTransaction);
+    });
+    // initialisation only
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  return (
+    <>
+      <DialogSection appearance={{ theme: 'sidePadding' }}>
+        <div className={styles.motionHeading}>
+          {isVotingExtensionEnabled && (
+            <MotionDomainSelect
+              colony={colony}
+              /*
+               * @NOTE Always disabled since you can only create this motion in root
+               */
+              disabled
+            />
+          )}
+          {canManageAndControlSafes && isVotingExtensionEnabled && (
+            <ForceToggle disabled={isSubmitting} />
+          )}
+        </div>
```

Comment:
I left it like this as I wasn't sure if there was a particular reason the `DialogHeading` component wasn't used in the Dapp. I have updated it to use the `DialogHeading` here b36d0364f7b58e9f3a6c7ef7ab927f6e86bf11ac but would appreciate a check to make sure I have got the permissions correct

Reply:
> I left it like this as I wasn't sure if there was a particular reason the DialogHeading component wasn't used in the Dapp.

That's okay, this component is actually an improvement that we made here in the CDapp so that's why it doesn't exist in the Dapp :)

I think you got it right except for the extra `canManageAndControlSafe` 👍

Reply:
hmmm interesting why dont we need the `canManageAndControlSafe` boolean now? i assumed this was necessary as an extra check that only applied to safes that fell outside the normal scope that the `DialogHeading` component dealt with